wall/wall.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * wall.c - write to all users
+ *
+ * Copyright (C) 2024: Luiz Antônio Rangel (takusuman)
+ *
+ * SPDX-Licence-Identifier: Zlib
+ *
+ *	from Unix 32V /usr/src/cmd/wall.c
+ *	December 19th, 1978.
+ *
+ * Copyright(C) Caldera International Inc. 2001-2002. All rights reserved.
+ *
+ * SPDX-Licence-Identifier: Caldera
+ */
+
+#include <errno.h>
+#include <limits.h>
+#include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strmenta.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <utmp.h>
+#include <utmpx.h>
+#define MAXNAMLEN _POSIX_LOGIN_NAME_MAX
+
+char	*mesg = "",
+	who[MAXNAMLEN] = "???",
+	sterm[32] = "";
+int	msize = 0;
+void main(int argc, char *argv[]);
+void sendmes(char *tty);
+
+void main(int argc, char *argv[]) {
+	int i = 0;
+	char c = '\0',
+	     smesg[3000] = "";
+	struct utmpx *utmp;
+	struct passwd *pw;
+	FILE *f = NULL;
+
+	if ((utmp = getutxent()) == NULL) {
+		fprintf(stderr, "failed to open utmpx database: %s\n",
+			strerror(errno));
+		exit(1);
+	}
+
+	f = stdin;
+	if(argc >= 2) {
+		/* take message from unix file instead of standard input */
+		if((f = fopen(argv[1], "r")) == NULL) {
+			fprintf(stderr, "Cannot open %s\n", argv[1]);
+			exit(1);
+		}
+	}
+	
+	for (; ((i = getc(f)) != EOF); ) smesg[msize++] = i;
+	fclose(f);
+	mesg = ssafe(smesg);
+	
+	pw = getpwuid(geteuid());
+	if (pw) {
+		for (i = 0; c = pw->pw_name[i]; i++) who[i]=c;
```

Comment:
Using `i` as both an iterating variable and a character is an horrible practise

Reply:
It's UNIX 7th code, they couldn't afford wasting memory.
May I change this or keep as it is? It works in practice.

Reply:
Also fixed. Using ``c`` instead of ``i`` since ``c`` was already defined.

Reply:
`int c` is usually used for characters to make it clear that it is a character and not a number

Reply:
> > It's UNIX 7th code, they couldn't afford wasting memory. May I change this or keep as it is? It works in practice.
> 
> Change this please, you can separate each part into a function if you think that your computer cant handle two integers

I've changed it, it's using ``c`` for characters now.